src/main/kotlin/no/nav/familie/ef/sak/task/BehandlingsstatistikkTask.kt

```diff
@@ -0,0 +1,151 @@
+import no.nav.familie.ef.sak.integration.dto.pdl.gjeldende
+import no.nav.familie.ef.sak.iverksett.IverksettClient
+import no.nav.familie.ef.sak.repository.domain.Fagsak
+import no.nav.familie.ef.sak.repository.domain.Stønadstype
+import no.nav.familie.ef.sak.service.BehandlingService
+import no.nav.familie.ef.sak.service.FagsakService
+import no.nav.familie.ef.sak.service.OppgaveService
+import no.nav.familie.ef.sak.service.PersonService
+import no.nav.familie.ef.sak.service.SøknadService
+import no.nav.familie.kontrakter.ef.felles.BehandlingType
+import no.nav.familie.kontrakter.ef.felles.StønadType
+import no.nav.familie.kontrakter.ef.iverksett.BehandlingsstatistikkDto
+import no.nav.familie.kontrakter.ef.iverksett.Hendelse
+import no.nav.familie.kontrakter.felles.oppgave.Oppgave
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import org.springframework.stereotype.Service
+import java.time.ZoneId
+import java.util.Properties
+import java.util.UUID
+
+@Service
+@TaskStepBeskrivelse(
+        taskStepType = BehandlingsstatistikkTask.TYPE,
+        beskrivelse = "Sender behandlingsstatistikk til iverksett"
+)
+
+class BehandlingsstatistikkTask(private val iverksettClient: IverksettClient,
+                                private val behandlingService: BehandlingService,
+                                private val fagsakService: FagsakService,
+                                private val søknadService: SøknadService,
+                                private val vedtakService: VedtakService,
+                                private val oppgaveService: OppgaveService,
+                                private val personService: PersonService
+) : AsyncTaskStep {
+    private val zoneIdOslo = ZoneId.of("Europe/Oslo")
+
+    override fun doTask(task: Task) {
+        val (behandlingId, hendelse, hendelseTidspunkt, gjeldendeSaksbehandler, oppgaveId) = objectMapper.readValue<BehandlingsstatistikkTaskPayload>(
+                task.payload
+        )
+
+        val personIdent = behandlingService.hentAktivIdent(behandlingId)
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        val fagsak = fagsakService.hentFagsak(behandling.fagsakId)
+
+        val sisteOppgaveForBehandling = finnSisteOppgaveForBehandlingen(behandlingId, oppgaveId)
+        val resultatBegrunnelse = finnResultatBegrunnelse(hendelse, behandlingId)
+        val søker = personService.hentSøker(personIdent);
+        val søknadstidspunkt = finnSøknadstidspunkt(fagsak, behandlingId)
+
+        val behandlingsstatistikkDto = BehandlingsstatistikkDto(
+                behandlingId = behandlingId,
+                personIdent = personIdent,
+                gjeldendeSaksbehandlerId = gjeldendeSaksbehandler,
+                eksternFagsakId = fagsak.eksternId.id.toString(),
+                hendelseTidspunkt = hendelseTidspunkt.atZone(zoneIdOslo),
```

Comment:
Burde vi bruke OffsetDateTime?

Reply:
Vet ikke, hvorfor?

Reply:
Det er vel egentligen en suppe, ble litt usikker nå. 
https://www.baeldung.com/java-zoneddatetime-offsetdatetime#differences

OffsetDateTime:
- should be used for storing a date in the database or communicating it over a network

Men, samtidig så er de tju ZonedDateTime som man skal bruke hvis man ønsker å forholde seg til normal/vintertid

Reply:
Jeg er ikke helt sikker på hva som blir mest riktig for oss å bruke. Så er ingen biggie

Det som er litt kul er denne, der man går miste om ZoneId 
```
fun main() {
    val message = objectMapper.writeValueAsString(mapOf("a" to ZonedDateTime.now(ZoneId.systemDefault())))
    println(message)
    val parsed = objectMapper.readValue<Map<String, ZonedDateTime>>(message)
    println(objectMapper.writeValueAsString(parsed))
}
{“a”:“2021-07-01T08:16:35.739818+02:00”}
{“a”:“2021-07-01T06:16:35.739818Z”}
```